ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/connections/QUICConnection_v1.kt

```diff
@@ -726,6 +762,89 @@ internal class QUICConnection_v1(
         private val logger = logger()
     }
 
+    private inner class StreamManager {
+        private val receiveStates = hashMapOf<Long, ReceiveStreamState>()
```

Comment:
note: Long will be boxed (not really an issue, just pointing)

also other note about synchronisation of this states, I mean, thread-safety
I don't see directly in code, if those can be used from multiple threads, so just pointing for additional attention.

Reply:
Yep, for now the goal was to make it work, it is definitely not the best approach. 

Multiple threads should not work with the same connection (at least for now)